zrml/swaps/src/lib.rs

```diff
@@ -250,6 +251,7 @@ mod pallet {
         /// * `max_pool_amount`: The calculated amount of assets for the pool must be equal or
         /// greater than the given value.
         #[pallet::weight(T::WeightInfo::pool_exit_with_exact_asset_amount())]
+        // MARK(non-transactional): Immediately calls and returns a transactional.
```

Comment:
I think ideally we would mark only dispatchables as transactional, since besides hooks those are the only entry points for state-changing functional calls in a pallet. In my opinion the dev should be responsible for ensuring the consistency of states when writing/modifying one of the state-changing entry point functions.

Reply:
Hmm, good point. Currently, some functions like `create_pool` are marked `transactional`. I'll open an issue for reviewing this design.